src/dnf-plugins/product-id/product-id.c

```diff
@@ -355,9 +350,8 @@ int pluginHook(PluginHandle *handle, PluginHookId id, DnfPluginHookData *hookDat
             free(repoProductId);
         }
 
-        freeProductDb(productDb);
         freeProductDb(oldProductDb);
-        g_ptr_array_unref(repos);
```

Comment:
We don't need to free `repos` anymore?  (My understanding of C memory management is rudimentary so forgive me if this is a dumb question)

Reply:
No. Decrementing reference on this object caused segfaults of PackageKit daemon. We are not "masters" of `repos` pointer here, because we do not create the `GPtrArray` using `g_ptr_array_sized_new()` as we do for other pointers, but we create `repos` using:

```c
GPtrArray *repos = dnf_context_get_repos(dnfContext);
```

There is no documentation about this function, but you was able to see some Glib critical error on stdout/stderr, when `microdnf` was used for testing. This error is gone and PackageKit daemon doesn't crash anymore, when `g_ptr_array_unref(repos);` was removed.

Reply:
Ah ha!